src_c/rwobject.c

```diff
@@ -398,6 +402,17 @@ _pg_rw_seek(SDL_RWops *context, Sint64 offset, int whence)
     Sint64 retval;
 #endif
 
+    if (helper->fileno != -1) {
+        if (!(offset == 0 &&
```

Comment:
Is the logic here correct?

It seems lseek could be called twice here in the case where offset != 0 && whence != SEEK_CUR, and lseek doesn't return -1?

Reply:
True. It did do the right thing but that extra lseek was unnecessary.